test/archethic/transaction_chain/transaction_data/recipients/arguments_encoding_test.exs

```diff
@@ -0,0 +1,30 @@
+defmodule Archethic.TransactionChain.TransactionData.Recipient.ArgumentsEncodingTest do
+  use ExUnit.Case
+  use ExUnitProperties
+
+  alias Archethic.TransactionChain.TransactionData.Recipient.ArgumentsEncoding
+
+  property "symmetric encoding/decoding of recipients arguments" do
+    check all(
+            args <-
+              StreamData.list_of(
+                StreamData.one_of([
```

Comment:
**Suggestion**
The recipient args can also have map, list anf float as value. It could be nice to add them too in the property testing.
But this already tested in the TypedEncodingTest, so it's not mandatory

Reply:
Yes this was the reason to not include complex types